adding-more-native-to-your-react-native-app.md

```diff
@@ -0,0 +1,186 @@
+- How to use it for iOS and Android platforms
+
+## What is it and what problems does it solve?
+
+First, let’s talk about ‘linking’ _outside_ of the context of React Native.
+
+Perhaps you may have thought it was related to `npm link` / `yarn link`; however, beyond both making dependencies whose
+source exists on a local file-system available to a project, these tools are meant to allow one to _work_ on those
+dependencies and take their name from ‘symbolic link’, the manner in which they are made available to an application. In
+case you were not thinking this–never mind me, let’s carry on.
+
+Instead, the origins lie with tooling thats exist in native tool-chains, such as those used for languages like `C`, the
+term ‘linking’ refers to building up a working executable from various compiled modules–referred to as ‘object files’,
+that contain the compiled ‘object code’ corresponding to input source files. These can either be ‘statically’ linked,
+which basically just copies over the ‘object code’ into the executable; or they can be ‘dynamically’ linked, meaning the
+‘object code’ is provided as a separate binary to the executable and loaded into memory at runtime. Why you would choose
+one over the other is beyond the scope of this presentation, but suffice it to say that for a native dependency’s code
+to be available to the application it needs to be linked in either of these manners.
+
+‘Linking’ in the context of React Native _refers_ to this process, but in reality only covers _configuring_ your project
+for the actual linker machinery to be able to do its work at build time. This is the part that the rest of this
+presentation is concerned about.
+
+## How was this done historically (for iOS)?
+
+Now that we understand what linking refers to in the context of React Native, let’s take a look at how this was done
+over the years in an iOS project.
+
+### Manual
+
+As taken from [the React Native documentation](https://facebook.github.io/react-native/docs/linking-libraries-ios#manual-linking),
+a native package is expected to come with a Xcode project that instructs Xcode how to build a static library out of the
+package’s native source.
+
+1. The user adds a reference to the package’s Xcode project to their own (app) Xcode project;
+1. after which they are able to select the package’s static library from the list of libraries their application should
+   link against;
+1. and finally, where necessary, update build settings of the app Xcode project required by the dependency, such as
+   including the correct C++ standard library.
+1. If, at runtime, you are only using the package through JavaScript then you are done now. If you also need to use the
+   package’s code from your own native code, you will need to setup header search paths so you can include the package’s
+   headers in your source files.
+1. If the package has any native dependencies of its own, you need to repeat the above steps for those dependencies.
```

Comment:
would be nice to show a diff of project.pbxproj after this step. 

Later show the diff (or lack thereof) of the new approach. That's a big contrast between manual/react-native link and autolink